cosmos/runtime/baseapp/baseapp.go

```diff
@@ -141,6 +148,33 @@ type PolarisBaseApp struct {
 	ERC20Keeper *erc20keeper.Keeper
 }
 
+// InitChainer initializes the chain.
+func (app *PolarisBaseApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) (abci.ResponseInitChain, error) {
+	// Validate that ethereum genesis and cosmos genesis timestamp are the same.
+	// TODO: investigate how we can make this work in evm/keeper/genesis.go
+	// currently doesn't work in there since the scope of gen is restricted to the evm component
+	var genesisState map[string]json.RawMessage
+	if err := json.Unmarshal(req.AppStateBytes, &genesisState); err != nil {
+		panic(err)
+	}
+
+	var ethGenesis *core.Genesis
+	if err := json.Unmarshal(genesisState["evm"], ethGenesis); err != nil {
+		return abci.ResponseInitChain{}, err
+	}
+
+	if ethGenesis.Timestamp != uint64(ctx.BlockHeader().Time.Unix()) {
```

Comment:
You already have the genesis file, so tbh I'd just do this without using the context to reduce uncertainty